cloudbuild.yaml

```diff
@@ -1,21 +1,61 @@
 # Provide instructions for google Cloud Build to auto-build flutter
 # dashboard to flutter-dashboard project. Auto-build will be triggered
-# by new commits to `master` branch.
+# by daily schedule on `master` branch.
+#
+# The auto-build will be skipped if no new commits since last deployment.
 
 steps:
+  # Obtain latest version.
+  - name: 'gcr.io/cloud-builders/gcloud'
+    entrypoint: '/bin/bash'
+    args:  
+      - '-c'
+      - |-
+        gcloud config set project $PROJECT_ID
+        latest_version=$(gcloud app versions list --hide-no-traffic --format 'value(version.id)')
+        # Save latest version to default domain `/workspace`, for use in other steps.
+        echo "$latest_version" > /workspace/latest_version.txt
+
   # Build app_flutter.
   - name: gcr.io/$PROJECT_ID/flutter
     entrypoint: '/bin/bash'
-    args: ['cloud_build/app_flutter_build.sh']
+    args: 
+      - '-c'
+      - |-
+        latest_version=$(cat /workspace/latest_version.txt)
+        current_version="version-$SHORT_SHA"
+        echo "latest_version: $latest_version"
+        echo "current_version: $current_version"
+        if [ "$latest_version" = "$current_version" ]; then
+          echo "No updates since last deployment."
+        else
+          bash cloud_build/app_flutter_build.sh
+        fi
 
-    # Build repo_dashboard.
+  #   # Build repo_dashboard.
   - name: gcr.io/$PROJECT_ID/flutter
     entrypoint: '/bin/bash'
-    args: ['cloud_build/repo_dash_build.sh']
+    args:  
+      - '-c'
+      - |-
+        latest_version=$(cat /workspace/latest_version.txt)
+        if [ "$latest_version" = "version-$SHORT_SHA" ]; then
+          echo "No updates since last deployment."
+        else
+          bash cloud_build/repo_dash_build.sh
+        fi
 
   # Deploy a new version to google cloud.
   - name: 'gcr.io/cloud-builders/gcloud'
     entrypoint: '/bin/bash'
-    args: ['cloud_build/deploy.sh', '$PROJECT_ID', '$SHORT_SHA', '$_GAE_PROMOTE']
+    args:  
+      - '-c'
```

Comment:
Couldn't we just add the check in the deploy step? e.g. build everything, but only deploy if it's a new version?

Reply:
I don't see the value to build everything we have already built/deployed before.

Reply:
Cocoon changes aren't the only thing that break our deploys, and it's helpful to know when those start to occur

Reply:
Ah, missed that part. Updated to always build.